### PR TITLE
git: Fix remote URL parsing for partial clones

### LINKED ISSUE #63025: editor: copy/open permalink to line fails with "remote "origin" not found" on all blobless clones
### Reproduction steps

1. Open Zed inside any repo that has been cloned using `--filter=blob:none`
2. Open any tracked file
3. Invoke `editor: copy permalink to line`

For example:
1. `git clone --filter=blob:none https://github.com/PostHog/posthog`
2. `cd posthog && zed .`
3. Open any tracked file; invoke `editor: copy permalink to line`

### Current vs. Expected behavior

A toast appears with an error:
`Failed to copy permalink: remote "origin" not found`

Image

I expect the permalink to be copied

### Zed version and system specs

Zed: 1.13.1 (stable)
OS: macOS (Apple Silicon)
Memory: 64 GiB
Architecture: aarch64

### Attach Zed log file

<details><summary>Zed.log</summary>

```log
2026-08-21T16:01:52+01:00 INFO  [git::repository] opening git repository at "/Users/ross/dev/posthog-tmp/.git" using git binary "/opt/homebrew/bin/git"
2026-08-21T16:01:52+01:00 WARN  [task::vscode_format] Skipping deserializing of a task `start-dev-with-debug` with the unsupported `dependsOn` key
2026-08-21T16:01:52+01:00 INFO  [project::context_server_store] posthog requires OAuth authorization (auth server: https://oauth.posthog.com/)
2026-08-21T16:02:00+01:00 ERROR [crates/editor/src/git.rs:1670] remote "origin" not found
2026-08-21T16:02:04+01:00 INFO  [project::lsp_store] language server 3: unregistered workspace/DidChangeWatchedFiles capability with id f9a62683-f3f1-4876-8a0e-7480957d4666
2026-08-21T16:02:04+01:00 INFO  [project::lsp_store] language server 3: unregistered workspace/DidChangeWatchedFiles capability with id 66519db8-eacb-418a-88cb-7f670b404a3b
2026-08-21T16:02:12+01:00 INFO  [zed::reliability] memory usage: resident 346 MiB (+87 MiB), virtual 585496 MiB
2026-08-21T16:02:12+01:00 INFO  [zed::reliability] worktree diagnostics: stores 2, slots 2, live 2, visible 2, strong 2, dead weak 0, loading 0, entries 109810, visible entries 106387, largest /Users/ross/dev/posthog (56891 entries, 53473 visible)

```

</details>

### Relevant Zed settings

<details><summary>settings.json</summary>

```json

```

</details>

### Relevant Keymap

<details><summary>keymap.json</summary>

```json

```

</details>

### (for AI issues) Model provider details

_No response_

### If you are using WSL on Windows, what flavor of Linux are you using?

None

### PR DESCRIPTION
**Pull request by rossgray**
Fixes #63025.

As a disclaimer, I do not have any experience with Rust, so had an agent help with this. Feel free to use this PR or re-implement the fix as you wish.

## Summary

Zed parses `git remote -v` output with `strip_suffix(" (fetch)")`. On promisor remotes (every `--filter=blob:none` clone, the default for `gh repo clone` on large repos), git appends a filter annotation like `[blob:none]` to the fetch line, which broke the suffix match and dropped that remote's URL from the map. Downstream, `editor: open permalink to line` then failed with `Failed to open permalink: remote "origin" not found`.

This strips the trailing `[filter]` annotation *before* the suffix check, so the parser handles both plain and annotated lines uniformly. URLs containing `[` are unaffected.

## Test plan

- [x] New regression test `test_get_remote_urls_with_partial_clone` — clones a repo with `--filter=blob:none` and asserts the remote appears in `remote_urls()`.
- [x] Full `cargo test -p git --lib` — 67/67 passing including pre-existing spaces-in-URL test.

## Checklist

- [x] I've read the [contributor guide](https://github.com/zed-industries/zed/blob/main/CONTRIBUTING.md)
- [x] My code follows Zed's style and lint rules (cargo fmt, clippy clean)

Release Notes:

- Fixed open/copy permalink for partially cloned git repositories

### CODE REVIEW

**Comment by rossgray** [comment]
@cla-bot check

File: crates/git/src/repository.rs [hunk 1]
```diff
@@ -1682,9 +1682,13 @@ impl GitRepository for RealGitRepository {
                 let mut urls = HashMap::default();
                 if let Ok(stdout) = git.run(&["remote", "-v"]).await {
                     for line in stdout.lines() {
-                        if let Some(line) = line.strip_suffix(" (fetch)")
-                            && let Some((name, url)) = line.split_once(char::is_whitespace)
-                        {
+                        // On promisor remotes, `git` appends `[filter]` (e.g. `[blob:none]`)
+                        // after `(fetch)`/`(push)`; strip it so the suffix check matches.
+                        let line = line.split(" [").next().unwrap_or(line);
```
Comment (yeskunall) [inline crates/git/src/repository.rs]: I’m a little sad that this is how we have to handle this (I personally don’t end up liking parsing+matching on strings, but that’s just me). Outside of that, it seems like this change is an understandable tradeoff of the change introduced in https://github.com/zed-industries/zed/pull/59053.

File: crates/git/src/repository.rs [hunk 2]
```diff
@@ -1682,9 +1682,13 @@ impl GitRepository for RealGitRepository {
                 let mut urls = HashMap::default();
                 if let Ok(stdout) = git.run(&["remote", "-v"]).await {
                     for line in stdout.lines() {
-                        if let Some(line) = line.strip_suffix(" (fetch)")
-                            && let Some((name, url)) = line.split_once(char::is_whitespace)
-                        {
+                        // On promisor remotes, `git` appends `[filter]` (e.g. `[blob:none]`)
```
Comment (yeskunall) [inline crates/git/src/repository.rs]: One concern: `split(" [")` treats the first occurrence of `" ["` anywhere in the line as the filter annotation, but that substring can be part of a valid remote URL or local path. When a remote configured as `/tmp/remote [archive].git` is rendered by Git as:

```text
origin  /tmp/remote [archive].git (fetch)
```

... this code truncates that to `origin  /tmp/remote`, after which the `(fetch)` check fails and `origin` is silently omitted.

Wondering if we instead parse from the right around the final ` (fetch)` marker and accept only either no remaining text or a trailing ` [filter]` annotation? That would handle ordinary and promisor remotes without altering the URL itself.

Comment (yeskunall) [reply crates/git/src/repository.rs]: It would also be good to add a regression case for a remote path containing `" ["`.

**Review by yeskunall — COMMENTED** [review]
Hey @rossgray! Great to see you contributing to Zed! Love the work ya’ll are doing over at PostHog. Wanna take a look at my comments and see if you agree with them? Even without Rust knowledge, the points I made should hopefully come across clearly, but LMK if I need to clarify something further.

File: crates/git/src/repository.rs (hunk 1 above)
Comment (rossgray) [reply crates/git/src/repository.rs]: Yes, I agree, it does certainly feel rather fragile doing things this way

File: crates/git/src/repository.rs (hunk 2 above)
Comment (rossgray) [reply crates/git/src/repository.rs]: Thanks for flagging this, it's an edge case I overlooked.

I've added a regression test for this (which I confirmed failed with my initial implementation) and updated the code to parse the line from the right as you suggested.

Let me know what you think

**Comment by smitbarmase** [comment]
Thanks for reporting this and putting up a fix. I hadn't seen this PR when I opened mine.

This version still misreads push URLs containing `(fetch)` because it ignores the text after the match. #63331 validates that text and includes regression tests for this case and the `[archive]` case from this PR.

Closing this in favor of that.
